.github/workflows/continuous-integration.yml

```diff
@@ -33,12 +34,15 @@ jobs:
         uses: "actions/cache@v1"
         with:
           path: "~/.composer/cache"
-          key: "php-${{ matrix.php-version }}-composer-locked-${{ hashFiles('composer.lock') }}"
+          key: "php-${{ matrix.php-version }}-composer-locked-${{ hashFiles('composer.lock') }}-${{ hashFiles('vendor-bin/phpunit/composer.lock') }}"
           restore-keys: "php-${{ matrix.php-version }}-composer-locked-"
 
       - name: "Install dependencies with composer"
         run: "composer install --no-interaction --no-progress --no-suggest"
 
+      - name: "Install toolchain with composer"
+        run: "composer bin phpunit install --no-interaction --no-progress --no-suggest"
```

Comment:
Why is phpunit in separate namespace from other tools? Do they not work together?

Reply:
That's because we run that job on 7.1, and psalm, phpstan and phpcs require higher versions of php. Note that all tools have their own namespace. It's just that here, I'm only installing phpunit.